### PR TITLE
fix(Tree): Drag and drop fix

### PR DESCRIPTION
**Pull request by leko91**

### CODE REVIEW

**Comment by SamuelAlev** [comment]
Fixes DnD issues caused by registration of items, now async items are correctly registered based on their index in JSX
```tsx
<Tree>
	<TreeItem> {/* sort: 0 */}
		<TreeItem /> {/* sort: 0 */}
		<TreeItem /> {/* sort: 1 */}
		<TreeItem /> {/* sort: 2 */}
	</TreeItem>
	<TreeItem /> {/* sort: 1 */}
	<TreeItem /> {/* sort: 2 */}
</Tree>
```

File: src/components/Tree/Tree.tsx [hunk 1]
```diff
@@ -64,40 +64,107 @@ const reducer = (state: TreeState, action: TreeStateAction): TreeState => {
...
-            }
-
             return {
                 ...state,
                 items: state.items.set(id, {
                     parentId,
-                    childrenIds,
                     level,
                     domElement: ref,
+                    childrenIds: undefined,
```
Comment (GoranRibic) [inline src/components/Tree/Tree.tsx:151]: why not pass an empty array? So you don't have to do `?` checks

File: src/components/Tree/TreeItem.tsx [hunk 2]
```diff
@@ -84,16 +80,17 @@ export const TreeItem = ({
...
     });
 
     const handleDrop = useCallback(
         (
-            targetItem: DraggableItem<{ id: string; sort: Nullable<number> }>,
-            sourceItem: DraggableItem<{ id: string; sort: Nullable<number> }>,
-            position: DropZonePosition,
+            targetItem: DraggableItem<{ id: string; sort: number }>,
+            sourceItem: DraggableItem<{ id: string; sort: number }>,
+            position: DropZonePosition, // [abc, def, ghi]
```
Comment (GoranRibic) [inline src/components/Tree/TreeItem.tsx:90]: // [abc, def, ghi] ?

File: src/components/Tree/TreeItem.tsx [hunk 3]
```diff
@@ -126,162 +123,165 @@ export const TreeItem = ({
...
-                    aria-level={treeItemState?.level}
-                    aria-selected={treeState.selectedIds.has(id)}
-                    aria-expanded={treeState.expandedIds.has(id)}
-                    tabIndex={0}
-                    onClick={handleSelect}
-                    onKeyDown={handleKeyDown}
-                    onMouseEnter={handleMouseEnter}
-                    onMouseLeave={handleMouseLeave}
+        <>
+            {(propsDraggable ?? draggable) && sort === 0 ? (
```
Comment (GoranRibic) [inline src/components/Tree/TreeItem.tsx:194]: what is propsDraggable?

**Review by GoranRibic — APPROVED** [review]

File: src/components/Tree/TreeItem.tsx (hunk 3 above)
Comment (SamuelAlev) [reply src/components/Tree/TreeItem.tsx:194]: `Add draggable on a per item basis.`
Allows remove dragging on one specific item (cover page and brandlink shouldn't be draggable for example)

File: src/components/Tree/Tree.tsx (hunk 1 above)
Comment (SamuelAlev) [reply src/components/Tree/Tree.tsx:151]: I prefer `undefined` for the leafs of a tree

File: src/components/Tree/TreeItem.tsx (hunk 3 above)
Comment (GoranRibic) [reply src/components/Tree/TreeItem.tsx]: I would give it a more meaningful name then, as it is confusing in my opinion. But that is not urgent ATM anyway. 👍

**Pull request merged**